src/repository/views.py

```diff
@@ -207,51 +207,61 @@ def repository_list(request, subject_slug=None):
     return render(request, template, context)
 
 
-# TODO: Re-implement
-def preprints_search(request, search_term=None):
+def repository_search(request, search_term=None):
     """
-    Searches through preprints based on their titles and authors
-    :param request: HttpRequest
-    :param search_term: Optional string
-    :return: HttpResponse
+    Searches for and displays a list of Preprints.
     """
+    if request.POST and 'search_term' in request.POST:
+        search_term = request.POST.get('search_term')
+        return redirect(
+            reverse(
+                'repository_search_with_term',
+                kwargs={'search_term': search_term},
+            )
+        )
+
+    # Grab all of the preprints that are published. We can then filter them
+    # if a search term is given or return them if none.
+    preprints = models.Preprint.objects.filter(
+        date_published__lte=timezone.now(),
+        repository=request.repository,
+    )
+
     if search_term:
         split_search_term = search_term.split(' ')
 
-        article_search = submission_models.Article.preprints.filter(
+        # Initial filter on Title, Abstract and Keywords.
+        preprint_search = preprints.filter(
             (Q(title__icontains=search_term) |
-             Q(subtitle__icontains=search_term) |
-             Q(keywords__word__in=split_search_term)),
-            stage=submission_models.STAGE_PREPRINT_PUBLISHED, date_published__lte=timezone.now()
+             Q(abstract__icontains=search_term) |
+             Q(keywords__word__in=split_search_term))
         )
-        article_search = [article for article in article_search]
 
-        institution_query = reduce(operator.and_, (Q(institution__icontains=x) for x in split_search_term))
-
-        from_author = core_models.Account.objects.filter(
-            (Q(first_name__in=split_search_term) |
-             Q(last_name__in=split_search_term) |
-             institution_query)
+        # Create a filter for author affiliation using the split term to check
+        # if each individual word is present.
+        affiliation_query = reduce(
+            operator.and_,
+            (Q(author__affiliation__icontains=x) for x in split_search_term)
```

Comment:
If I search for "University of Never-land" this is likely to return results for preprints from authors affiliated with:
    "Trump University",
    "League of Justice"

Makes me wonder if the split term search would be at all useful (on top of the search overhead for search terms with many words)

It does make more sense to keep it for author names and keywords though